test_runner/regress/test_ondemand_download.py

```diff
@@ -357,8 +357,23 @@ def get_resident_physical_size():
             tenant_id, timeline_id, "pageserver_resident_physical_size"
         )
 
+    # Shut down safekeepers before starting the pageserver.
```

Comment:
May be it is better to introduce PS parameter for delaying logical size computation?
Itmay be useful not only for tests...

Reply:
I don't think so. Problem here is not the fact that safekeeper communication ends up asking logical size, that's what the failpoint is for, set as environment variable when starting up pageserver.

Problem was that new WAL was being received in apparently when there's other load on the test runners leading to flakyness of the test, and my previous "fix" made this more visible.

I should look at the comments because as noted in slack, at best they are distracting.

Reply:
Cleaned the test up in e249c8a9a not to allude to safekeepers launching logical size calculation.

Reply:
Checked the example I linked here https://github.com/neondatabase/neon/issues/3831#issuecomment-1721217426:

> EDIT: Log analysis later on: logs clearly show inmemory flush after metrics were read during shutdown